operators/src/pro/cache/shared_cache.rs

```diff
@@ -0,0 +1,1190 @@
+            );
+
+            Some(entry)
+        } else {
+            None
+        }
+    }
+
+    /// This method removes a query from the cache and the LRU.
+    /// It will remove a queries cache entry from the cache and the LRU.
+    ///
+    /// If the query is not in the cache, this method returns None.
+    ///
+    fn remove_query_from_cache_and_lru(
+        &mut self,
+        cache_entry_id: &CacheEntryId,
+    ) -> Option<CacheQueryEntry<C::Query, C::CacheContainer>> {
+        if let Some(entry) = self.operator_cache.remove_cache_entry(cache_entry_id) {
+            let old_lru_entry = self.lru.pop_entry(cache_entry_id);
+            debug_assert!(old_lru_entry.is_some(), "CacheEntryId not found in LRU");
+            self.cache_size.remove_element_bytes(cache_entry_id);
+            self.cache_size.remove_element_bytes(&entry);
+
+            log::debug!(
+                "Removed cache entry {}. Cache size: {}. Cache size used: {}, Cache used percentage: {}.",
+                cache_entry_id, self.cache_size.total_byte_size(), self.cache_size.byte_size_used(), self.cache_size.size_used_fraction()
+            );
+
+            Some(entry)
+        } else {
+            None
+        }
+    }
+
+    /// This method removes a list of queries from the cache and the LRU.
+    fn discard_querys_from_cache_and_lru(&mut self, cache_entry_ids: &[CacheEntryId]) {
+        for cache_entry_id in cache_entry_ids {
+            let old_entry = self.remove_query_from_cache_and_lru(cache_entry_id);
+            debug_assert!(
+                old_entry.is_some(),
+                "CacheEntryId not found in OperatorCacheEntry"
+            );
+        }
+    }
+
+    /// This method adds a query element to the landing zone.
+    /// It will add the element to the landing zone entry of the query.
+    ///
+    /// # Errors
+    ///
+    /// This method returns an error if the query is not in the landing zone.
+    /// This method returns an error if the element is already expired.
+    /// This method returns an error if the landing zone is full or the new element would cause the landing zone to overflow.
+    ///
+    fn add_query_element_to_landing_zone(
+        &mut self,
+        query_id: &QueryId,
+        landing_zone_element: C,
+    ) -> Result<(), CacheError> {
+        let landing_zone_entry = self
+            .operator_cache
+            .landing_zone_entry_mut(query_id)
+            .ok_or(CacheError::QueryNotFoundInLandingZone)?;
+
+        if landing_zone_element.cache_hint().is_expired() {
+            log::trace!("Element is already expired");
+            return Err(CacheError::TileExpiredBeforeInsertion);
+        };
+
+        let element_bytes_size = landing_zone_element.byte_size();
+
+        if !self.landing_zone_size.can_fit_bytes(element_bytes_size) {
+            return Err(CacheError::NotEnoughSpaceInLandingZone);
```

Comment:
where do you remove the landing_zone_entry in this case?

Reply:
ah, I see it's line 470 right?

Reply:
yes